cmd/crls-poc/main.go

```diff
@@ -0,0 +1,196 @@
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/x509"
+	"flag"
+	"io"
+	"math/big"
+	mrand "math/rand"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/honeycombio/beeline-go"
+	"google.golang.org/grpc/health"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+
+	"github.com/letsencrypt/boulder/cmd"
+	crlpb "github.com/letsencrypt/boulder/cmd/crls-poc/proto"
+	"github.com/letsencrypt/boulder/cmd/crls-poc/server"
+	"github.com/letsencrypt/boulder/core"
+	"github.com/letsencrypt/boulder/features"
+	bgrpc "github.com/letsencrypt/boulder/grpc"
+	"github.com/letsencrypt/boulder/issuance"
+)
+
+type Config struct {
+	Common struct {
+		cmd.ServiceConfig
+		Features map[string]bool
+	}
+
+	Server struct {
+		GRPCCRLGenerator *cmd.GRPCServerConfig
+
+		// Issuers is a list of all issuers which can sign CRLs.
+		Issuers []issuance.IssuerConfig
+
+		// LifespanCRL is how long CRLs are valid for. Per the BRs, Section 4.9.7,
+		// it MUST NOT be more than 10 days.
+		LifespanCRL cmd.ConfigDuration
```

Comment:
Is there an internal check for this that will cause the crl generator to fail to start if an sre attempts to deploy a lifespan tgat violates the BRs?

Reply:
```
	if lifetime >= 10*24*time.Hour {
		return nil, fmt.Errorf("crl lifetime cannot be more than 10 days, got: %s", lifetime)
	}
```
in crl.go

Reply:
Thanks, I missed that.